File: classes/parte-8-como-funcionan-los-lenguajes/134-tareas-corrutinas-y-canales/implementaciones/c/main.c
#include <stdio.h>

int main(void) {
    long x, maximo;
    if (scanf("%ld", &maximo) != 1) return 1;
    while (scanf("%ld", &x) == 1) {
        if (x > maximo) maximo = x;
    }
    printf("max=%ld\n", maximo);
    return 0;
}

File: classes/parte-8-como-funcionan-los-lenguajes/134-tareas-corrutinas-y-canales/implementaciones/go/main.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"strings"
)

func main() {
	line, _ := bufio.NewReader(os.Stdin).ReadString('\n')
	f := strings.Fields(line)
	ch := make(chan int, len(f))
	go func() { // productor
		for _, s := range f {
			n, _ := strconv.Atoi(s)
			ch <- n
		}
		close(ch)
	}()
	primero := true
	maximo := 0
	for x := range ch { // consumidor
		if primero || x > maximo {
			maximo = x
			primero = false
		}
	}
	fmt.Printf("max=%d\n", maximo)
}
